meson.build

```
@@ -0,0 +1,106 @@
+project('nntrainer', 'c', 'cpp',
+  version: '0.0.1',
+  license: ['apache-2.0'],
+  meson_version: '>=0.50.0',
+  default_options: [
+    'werror=true',
+    'warning_level=1',
+    'c_std=gnu89',
+    'cpp_std=c++14'
+  ]
+)
+
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+build_platform = ''
+
+if get_option('enable-tizen')
+  # Pass __TIZEN__ to the compiler
+  build_platform = 'tizen'
+endif
+
+warning_flags = [
+  '-Wredundant-decls',
+  '-Wwrite-strings',
+  '-Wformat',
+  '-Wformat-nonliteral',
+  '-Wformat-security',
+  '-Winit-self',
+  '-Waddress',
+  '-Wno-multichar',
+  '-Wvla',
+  '-Wpointer-arith'
+]
+
+warning_c_flags = [
+  '-Wmissing-declarations',
+  '-Wmissing-include-dirs',
+  '-Wmissing-prototypes',
+  '-Wnested-externs',
+  '-Waggregate-return',
+  '-Wold-style-definition',
+  '-Wdeclaration-after-statement'
+]
+
+foreach extra_arg : warning_flags
+  if cc.has_argument (extra_arg)
+    add_project_arguments([extra_arg], language: 'c')
+  endif
+  if cxx.has_argument (extra_arg)
+    add_project_arguments([extra_arg], language: 'cpp')
+  endif
+endforeach
+
+foreach extra_arg : warning_c_flags
+  if cc.has_argument (extra_arg)
+    add_project_arguments([extra_arg], language: 'c')
+  endif
+endforeach
+
+# Set install path
+nntrainer_prefix = get_option('prefix')
+nntrainer_libdir = join_paths(nntrainer_prefix, get_option('libdir'))
+nntrainer_bindir = join_paths(nntrainer_prefix, get_option('bindir'))
+nntrainer_includedir = join_paths(nntrainer_prefix, get_option('includedir'))
+nntrainer_inidir = get_option('sysconfdir')
+application_install_dir = join_paths(nntrainer_bindir, 'applications')
+
+
+# Set default configuration
+nntrainer_conf = configuration_data()
+nntrainer_conf.set('VERSION', meson.project_version())
+nntrainer_conf.set('PREFIX', nntrainer_prefix)
+nntrainer_conf.set('EXEC_PREFIX', nntrainer_bindir)
+nntrainer_conf.set('LIB_INSTALL_DIR', nntrainer_libdir)
+nntrainer_conf.set('INCLUDE_INSTALL_DIR', nntrainer_includedir)
+
+# Dependencies
+if get_option('enable-blas')
+  add_project_arguments('-DUSE_BLAS=1', language:['c','cpp'])
+  if build_platform == 'tizen'
+    blas_dep = dependency('openblas')
+  else
+    blas_dep = dependency('blas-openblas')
+  endif
+endif
+
+libm_dep = cc.find_library('m') # cmath library
+libdl_dep = cc.find_library('dl') # DL library
```

```suggestion
libdl_dep = cxx.find_library('dl') # DL library
```

It seems that there are no c source files.

---

Fixed!